test/indexer/test_hca_indexer.py

```diff
@@ -1256,6 +1259,45 @@ def test_accessions_fields(self):
             self.assertEqual(['E-AAAA-00'], project['array_express_accessions'])
             self.assertEqual(['PRJNA000000'], project['insdc_study_accessions'])
 
+    def test_cell_counts(self):
+        """
+        Verify the cell counts found in project, cell_suspension, and file entities
+        """
+        # Bundles from the canned staging area, both for project 90bf705c
+        # https://github.com/HumanCellAtlas/schema-test-data/
+        bundle_fqid = self.bundle_fqid(uuid='4da04038-adab-59a9-b6c4-3a61242cc972',
+                                       version='2021-01-01T00:00:00.000000Z')
+        self._index_canned_bundle(bundle_fqid)
+        bundle_fqid = self.bundle_fqid(uuid='d7b8cbff-aee9-5a05-a4a1-d8f4e720aee7',
+                                       version='2021-01-01T00:00:00.000000Z')
+        self._index_canned_bundle(bundle_fqid)
+        hits = self._get_all_hits()
+
+        field_paths = [
+            ('projects', 'estimated_cell_count'),
+            ('cell_suspensions', 'total_estimated_cells'),
+            ('files', 'matrix_cell_count')
+        ]
+        actual = NestedDict(2, int)
+        for hit in sorted(hits, key=lambda d: d['_id']):
+            entity_type, aggregate = self._parse_index_name(hit)
+            contents = hit['_source']['contents']
+            for inner_entity_type, field_name in field_paths:
+                for inner_entity in contents[inner_entity_type]:
+                    value = inner_entity[field_name]
+                    actual[aggregate][entity_type][inner_entity_type] += value
+
+        expected = NestedDict(1, dict)
+        for aggregate in False, True:
+            for entity_type in self.index_service.entity_types(self.catalog):
+                expected[aggregate][entity_type] = {
+                    'projects': 10000 if aggregate and entity_type == 'projects' else 20000,
```

Comment:
Re https://github.com/DataBiosphere/azul/pull/3361#discussion_r722397769

Ahh, so the leaves of the fingerprint should be lists

```diff
Index: test/indexer/test_hca_indexer.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/indexer/test_hca_indexer.py b/test/indexer/test_hca_indexer.py
--- a/test/indexer/test_hca_indexer.py	(revision 3e164ebd69e9a0609fca603750bd23d3bfbadda2)
+++ b/test/indexer/test_hca_indexer.py	(date 1633480153563)
@@ -1,3 +1,4 @@
+from bisect import insort
 from collections import (
     Counter,
     defaultdict,
@@ -1278,22 +1279,24 @@
             ('cell_suspensions', 'total_estimated_cells'),
             ('files', 'matrix_cell_count')
         ]
-        actual = NestedDict(2, int)
+        actual = NestedDict(2, list)
         for hit in sorted(hits, key=lambda d: d['_id']):
             entity_type, aggregate = self._parse_index_name(hit)
             contents = hit['_source']['contents']
             for inner_entity_type, field_name in field_paths:
                 for inner_entity in contents[inner_entity_type]:
                     value = inner_entity[field_name]
-                    actual[aggregate][entity_type][inner_entity_type] += value
+                    insort(actual[aggregate][entity_type][inner_entity_type], value)
 
         expected = NestedDict(1, dict)
         for aggregate in False, True:
             for entity_type in self.index_service.entity_types(self.catalog):
+                is_project_aggregate = aggregate and entity_type == 'projects'
                 expected[aggregate][entity_type] = {
-                    'projects': 10000 if aggregate and entity_type == 'projects' else 20000,
-                    'cell_suspensions': 40000,
-                    'files': 17100
+                    # estimated_cell_count is aggregated using max, not sum
+                    'projects': [10000] if is_project_aggregate else [10000, 10000],
+                    'cell_suspensions': [40000] if is_project_aggregate else [20000, 20000],
+                    'files': [17100] if is_project_aggregate else [2100, 15000]
                 }
 
         self.assertEqual(expected.to_dict(), actual.to_dict())
```